src/frontend/mame/luaengine.cpp

```diff
@@ -987,6 +987,38 @@ void lua_engine::initialize()
 			return table;
 		}));
 
+/*  emu_options library
+ *
+ * manager:options()
+ * manager:machine():options()
+ *
+ * options:slot_option() - retrieves a specific slot option
+ *
+ * options.entries[] - get table of option entries (k=name, v=core_options::entry)
+ */
+
+	auto emu_options_type = sol().registry().new_usertype<emu_options>("new", sol::no_constructor, sol::base_classes, sol::bases<core_options>());
+	emu_options_type.set("slot_option", [](emu_options &opts, const std::string &name) { return opts.find_slot_option(name); });
+	sol().registry().set("emu_options", emu_options_type);
+
+/*  slot_option library
+ *
+ * manager:options():slot_option("name")
+ * manager:machine():options():slot_option("name")
+ *
+ * slot_option:value()
+ * slot_option:specified_value()
+ * slot_option:bios()
+ * slot_option:default_card_software()
+ */
+
+	auto slot_option_type = sol().registry().new_usertype<slot_option>("new", sol::no_constructor);
+	slot_option_type.set("value", &slot_option::value);
+	slot_option_type.set("specified_value", &slot_option::specified_value);
+	slot_option_type.set("bios", &slot_option::bios);
+	slot_option_type.set("default_card_software", &slot_option::default_card_software);
+	slot_option_type.set("specify", [](slot_option &opt, std::string &&text) { opt.specify(std::move(text)); });
```

Comment:
You haven’t documented this at all.

Reply:
Should this be a method, or should the `value` property be writable?  Maybe it needs to be a method because of the BIOS option.  But speaking of the BIOS option, maybe it should take two arguments, and if the second argument is not supplied or is Lua `nil` not put the `,bios=` part in.  We don’t really need to expose that particular piece of ugliness to Lua.

Also, for all the class wrappers we’ve been putting methods before properties, so if this remains a method, it should come before the properties for consistency.